### PR TITLE
Add mutation function to applications.go

### PR DESCRIPTION
**Pull request by shaior**
Added WithGitDetails mutation function, this is needed in order to define the argocd app with git details.

### CODE REVIEW

File: pkg/argocd/applications.go [hunk 1]
```diff
@@ -202,3 +202,23 @@ func (builder *ApplicationBuilder) validate() (bool, error) {
 
 	return true, nil
 }
+
+// WithGitDetails applies git details to application definition.
+func (builder *ApplicationBuilder) WithGitDetails(gitRepo, gitBranch, gitPath string) *ApplicationBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
```
Comment (kononovn) [inline pkg/argocd/applications.go:211]: Please add argument verification. 
You need to make sure that arguments are not empty strings

**Review by kononovn — CHANGES_REQUESTED** [review]
Please fix comment

File: pkg/argocd/applications.go (hunk 1 above)
Comment (shaior) [reply pkg/argocd/applications.go:211]: Done, thanks.

File: pkg/argocd/applications.go [hunk 2]
```diff
@@ -202,3 +202,41 @@ func (builder *ApplicationBuilder) validate() (bool, error) {
...
+		builder.errorMsg = "'gitBranch' parameter is empty"
+	}
+
+	if gitPath == "" {
+		glog.V(100).Infof("The 'gitPath' of the argocd application is empty")
+
+		builder.errorMsg = "'gitPath' parameter is empty"
+	}
+
+	glog.V(100).Infof("Redefining argocd application %s in namespace: %s with git details",
```
Comment (kononovn) [inline pkg/argocd/applications.go:230]: Please add parameters value to log message

**Review by kononovn — CHANGES_REQUESTED** [review]
1 more comment

File: pkg/argocd/applications.go (hunk 2 above)
Comment (achuzhoy) [inline pkg/argocd/applications.go:230]: inconsistent use of colon, i.e.no colon after application and yes colon after namespace

**Review by kononovn — APPROVED** [review]
lgtm

File: pkg/argocd/applications.go [hunk 3]
```diff
@@ -202,3 +202,44 @@ func (builder *ApplicationBuilder) validate() (bool, error) {
...
+	}
+
+	if gitPath == "" {
+		glog.V(100).Infof("The 'gitPath' of the argocd application is empty")
+
+		builder.errorMsg = "'gitPath' parameter is empty"
+	}
+
+	glog.V(100).Infof(
+		"Redefining argocd application: %s in namespace: %s with the following git details "+
```
Comment (trewest) [inline pkg/argocd/applications.go:231]: You're not neccessarilty redefining the application, so I think it's better to change the wording to the following
```suggestion
		"Adding the following git details to the argocd application: %s in namespace: %s "+
```

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**